object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/KnnPredicate.java

```diff
@@ -40,12 +40,13 @@ public ValueExpr getChild() {
 
    @Override
    public String toString() {
-      return leftChild.toString() + "<-> " + vector + "~" + knn;
+      return toQueryString();
    }
 
    @Override
    public String toQueryString() {
-      return leftChild.toQueryString() + "<-> " + vector + "~" + knn;
+      String printedVector = (vectorParam == null) ? vector + "" : "[" + vectorParam + "]";
+      return leftChild.toQueryString() + "<-> " + printedVector + "~" + knn;
```

Comment:
How deep can these things be? I wonder if we should have an optimized toQueryString that writes to a StringBuilder instead.

Reply:
open AI is 1000

Reply:
Replacing with a String builder looks a good idea. I applied it.